diff --git a/src/shared/missing.h b/src/shared/missing.h
index 5ad5997..784725b 100644
--- a/src/shared/missing.h
+++ b/src/shared/missing.h
@@ -129,7 +129,7 @@ static inline pid_t gettid(void) {
 #  elif defined(__powerpc__)
 #    define __NR_name_to_handle_at 345
 #  else
-#    error "__NR_name_to_handle_at is not defined"
+#   warning "__NR_name_to_handle_at is not defined"
 #  endif
 #endif
 
@@ -140,10 +140,17 @@ struct file_handle {
         unsigned char f_handle[0];
 };
 
+#ifndef __NR_name_to_handle_at
+static inline int name_to_handle_at(int fd, const char *name, struct file_handle *handle, int *mnt_id, int flags) {
+        errno = ENOSYS;
+        return -1;
+}
+#else
 static inline int name_to_handle_at(int fd, const char *name, struct file_handle *handle, int *mnt_id, int flags) {
         return syscall(__NR_name_to_handle_at, fd, name, handle, mnt_id, flags);
 }
 #endif
+#endif
 
 #ifndef INPUT_PROP_POINTING_STICK
 #define INPUT_PROP_POINTING_STICK 0x05
